src/Compilers/VisualBasic/Test/IOperation/IOperation/IOperationTests_IAwaitExpression.vb

```diff
@@ -211,5 +211,78 @@
 
             VerifyOperationTreeAndDiagnosticsForTest(Of ExpressionStatementSyntax)(source, expectedOperationTree, expectedDiagnostics, useLatestFramework:=True)
         End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/67616")>
+        Public Sub TestAwaitExpression_InStatement()
+            Dim source = <![CDATA[
+Imports System.Threading.Tasks
+
+Public Module Program
+    Public Async Function M() As Task(Of Integer)
+        Await M2()'BIND:"Await M2()"
+        Return 0
+    End Function
+
+    Public Function M2() As Task(Of String)
+        Throw New System.Exception()
+    End Function
+End Module
+]]>.Value
+
+            Dim expectedOperationTree = <![CDATA[
+IAwaitOperation (OperationKind.Await, Type: System.String) (Syntax: 'Await M2()')
+  Expression:
+    IInvocationOperation (Function Program.M2() As System.Threading.Tasks.Task(Of System.String)) (OperationKind.Invocation, Type: System.Threading.Tasks.Task(Of System.String)) (Syntax: 'M2()')
+      Instance Receiver:
+        null
+      Arguments(0)
+]]>.Value
+
+            Dim expectedDiagnostics = String.Empty
+
+            VerifyOperationTreeAndDiagnosticsForTest(Of AwaitExpressionSyntax)(source, expectedOperationTree, expectedDiagnostics, useLatestFramework:=True)
+        End Sub
+
+        <CompilerTrait(CompilerFeature.IOperation)>
+        <Fact, WorkItem("https://github.com/dotnet/roslyn/issues/67616")>
+        Public Sub TestAwaitExpression_InStatement_InSubLambda()
+            Dim source = <![CDATA[
+Imports System
+Imports System.Threading.Tasks
+
+Public Module Program
+    Public Async Function M() as Task
+        Dim lambda As Action = Async Sub()
+                                 Await M2()'BIND:"Await M2()"
+                               End Sub
```

Comment:
This isn't an expression lambda though. And expression lambda would look like 
```
Async Sub() Await M2()
```
Let's keep this scenario too.

Reply:
> Let's keep this scenario too.

Here I was referring to the body of the lambda rather than to the error.